Update SecretStore API documentation to account for Features feature. (#2863)
dapr/docs: daprdocs/content/en/reference/api/secrets_api.md

#### Response Body

If a secret store has support for multiple key-values in a secret, a JSON payload is returned with the key names as fields and their respective values.

In case of a secret store that only has name/value semantics, a JSON payload is returned with the name of the secret as the field and the value of the secret as the value.

> [The list of Supported Secret Stores]({{< ref supported-secret-stores.md >}}) provides a classification of secret stores that support multiple keys in a secret and those with name/value semantics.

Reviewer: ```suggestion
[See the classification of secret stores]({{< ref supported-secret-stores.md >}}) that support multiple keys in a secret and name/value semantics.
```

}
```

> Notice the name of the secret (`db-secret`) is **not** returned as part of the result. Contrast this with the result from a secret store with name/value semantics as shown bellow.

Reviewer: ```suggestion
The above example demonstrates a response from a secret store with multiple keys in a secret. Note that the secret name (`db-secret`) **is not** returned as part of the result. 
```

> Notice here that the name of the secret (`db-secret`) is **not** returned as part of the result. Contrast this with the result from a secret store with name/value semantics as shown bellow.

Reviewer 2: ```suggestion
Notice the name of the secret (`db-secret`) is **not** returned as part of the result. Contrast this with the result from a secret store with name/value semantics as shown bellow.
```

}
```

> The result has a single key-value pair and the name of the secret (`db-secret`) **is** returned as the key in that key-value pair.

Reviewer: ```suggestion
The above example demonstrates a response from a secret store with name/value semantics. Compared to the result from a secret store with multiple keys in a secret, this result returns a single key-value pair, with the secret name (`db-secret`) returned as the key in the key-value pair.
```

> Observe that the result has a single key-value pair and that name of the secret (`db-secret`) **is** returned as the key in that key-value pair.

Reviewer 2: ```suggestion
The result has a single key-value pair and the name of the secret (`db-secret`) **is** returned as the key in that key-value pair.
```
